scripts/production/homology_stats_rapid.py

```diff
@@ -0,0 +1,189 @@
+"""
+
+import argparse
+import json
+from typing import Tuple, List
+
+from sqlalchemy import create_engine, text
+from sqlalchemy.engine import Connection
+from ensembl.database import DBConnection
+from ensembl.ncbi_taxonomy.api.utils import Taxonomy
+
+
+def parse_arguments() -> argparse.Namespace:
+    """
+    Parse command line arguments.
+
+    Returns:
+        argparse.Namespace: Parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description="Query gene and homology counts from an Ensembl compara rapid release database."
+    )
+    parser.add_argument("-d", "--database", required=True, help="Database URL")
+    parser.add_argument(
+        "-r",
+        "--ref_db",
+        required=True,
+        help="Reference database name. Must be on same server as target database.",
+    )
+    parser.add_argument(
+        "-o", "--output", required=True, help="Output file in JSON format"
+    )
+    return parser.parse_args()
+
+
+def get_closest_ref(
+    tax_dbc: DBConnection, rr_dbc: Connection, ref_db: str
+) -> Tuple[int, str, int, str, int]:
+    """
+    Get the closest reference for the given rapid release database.
+
+    Args:
+        tax_dbc (DBConnection): Taxonomy database connection
+        rr_dbc (Connection): Rapid release database connection
+        ref_db (str): Reference database name
+
+    Returns:
+        Tuple[int, str, int, str, int]: Result tuple
+    """
+    query1 = """
+    SELECT name, taxon_id FROM genome_db LIMIT 1;
+    """
+
+    with rr_dbc.connect() as conn:
+        tmp = conn.execute(text(query1)).one()
+        query_species, query_taxid = tmp["name"], tmp["taxon_id"]
+
+    query2 = f"""
+    SELECT genome_db_id, gdb.taxon_id, gdb.name FROM
+    species_set JOIN {ref_db}.genome_db gdb USING(genome_db_id)
+    WHERE genome_db_id > 100;
+    """
+
+    with rr_dbc.connect() as conn, tax_dbc.session_scope() as session:
+        result = conn.execute(text(query2))
+        res: List[Tuple] = []
+        for dict_row in result.mappings():
+            taxid = dict_row["taxon_id"]
+            if taxid == query_taxid:
+                continue
+            anc = Taxonomy.all_common_ancestors(session, query_taxid, taxid)
+            res.append((dict_row["genome_db_id"], dict_row["name"], taxid, len(anc)))
+    res = sorted(res, key=lambda x: (x[3], x[2]), reverse=True)
+    return res[0][0], res[0][1], res[0][2], query_species, query_taxid
+
+
+def query_rr_database(
+    rr_dbc: Connection, genome_db_id: int, ref_db: str
+) -> Tuple[int, int]:
+    """
+    Query the database for the number of homologs and genes.
+
+    Args:
+        rr_dbc (Connection): Rapid release database connection
+        genome_db_id (int): Genome database ID
+        ref_db (str): Reference database name
+
+    Returns:
+        Tuple[int, int]: Tuple containing the number of homologs and genes
+    """
+    connection: Connection = rr_dbc.connect()
```

Comment:
Would it be possible to open this connection in a `with` block encompassing the code to get `nr_homologs` and `nr_genes`? 
```suggestion
    with rr_dbc.connect() as connection:
```